### DISCUSSION TITLE
Pushing my commits aren't going to my forked repo

### DISCUSSION THREAD

**Discussion by rfalanga** [community/community#23182] [New to GitHub]
<p>I am having difficulties pushing changes I’ve made to a forked a repo I made.</p>
<p>I wanted to make some changes to the README.md file for my local .NET user group GitHub repo. I forked the repo into a copy of the original under my GitHub account. Since I’m new at this I looked at <a href="https://app.egghead.io/lessons/javascript-how-to-fork-and-clone-a-github-repository" rel="noopener nofollow ugc">Kent Dodds tutorial</a>) on how to configure things so my local repo would fetch from the original repo, but I had thought push to my forked repo. Or at least that’s what I thought.</p>
<p>But I’ve made some mistake. After I made edits and committed them locally, I tried to push my changes to my forked repo. However, this is the error I got:</p>
<blockquote>
<p>D:\GitHub\<a href="http://nmug.github.io" rel="noopener nofollow ugc">nmug.github.io</a> [master ↑1]&gt; git push<br>
remote: Permission to nmug/nmug.github.io.git denied to rfalanga.<br>
fatal: unable to access ‘<a href="https://github.com/nmug/nmug.github.io.git/" rel="noopener nofollow ugc">https://github.com/nmug/nmug.github.io.git/</a>’: The requested URL returned error: 403</p>
</blockquote>
<p>So next I entered checked my remotes and here’s what I got:</p>
<blockquote>
<p>D:\GitHub\<a href="http://nmug.github.io" rel="noopener nofollow ugc">nmug.github.io</a> [master ↑1]&gt; git remote -v<br>
origin  <a href="https://github.com/rfalanga/nmug.github.io.git" rel="noopener nofollow ugc">https://github.com/rfalanga/nmug.github.io.git</a> (fetch)<br>
origin  <a href="https://github.com/rfalanga/nmug.github.io.git" rel="noopener nofollow ugc">https://github.com/rfalanga/nmug.github.io.git</a> (push)<br>
upstream        <a href="https://github.com/nmug/nmug.github.io.git" rel="noopener nofollow ugc">https://github.com/nmug/nmug.github.io.git</a> (fetch)<br>
upstream        <a href="https://github.com/nmug/nmug.github.io.git" rel="noopener nofollow ugc">https://github.com/nmug/nmug.github.io.git</a> (push)</p>
</blockquote>
<p>I want to fetch/pull from upstream, but push to my origin. What have I done wrong and how do I fix it, please?</p>
<p>(I’m using posh-git)</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>If you want <code>git pull</code> without arguments to keep pulling from <code>upstream</code> you’ll need to explicitly <code>push</code> to your <code>origin</code>:</p>
<pre><code class="lang-bash">git push origin master
</code></pre>
<p>Otherwise you can set the upstream branch, but after that you’d need to explicitly <em>pull</em> from <code>upstream</code>:</p>
<pre><code class="lang-bash">git branch --set-upstream origin/master
</code></pre>

**Comment by rfalanga** [comment]
<p>Unfortunately, the <code>git push origin master</code> command didn’t work.  After I entered it, then tried to push to my forked branch I still got the same error:</p>
<blockquote>
<p>D:\GitHub\<a href="http://nmug.github.io" rel="noopener nofollow ugc">nmug.github.io</a> [master ↑1]&gt; git push<br>
remote: Permission to nmug/nmug.github.io.git denied to rfalanga.<br>
fatal: unable to access ‘<a href="https://github.com/nmug/nmug.github.io.git/" rel="noopener nofollow ugc">https://github.com/nmug/nmug.github.io.git/</a>’: The requested URL returned error: 403</p>
</blockquote>
<p>I’ll try setting the upstream branch.</p>

**Comment by rfalanga** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a>, I had to change <code>--set-upstream</code> to <code>--set-upstream-to</code> as the former is no longer supported.</p>
<p>I was able to push my changes to my forked repo. Then I created a PR for the original author to consider.</p>
<p>Thank you for your help.</p>
